internal/static/cartogram2.js

```diff
@@ -39,21 +39,55 @@ function clearFileInput(ctrl) {
     if (ctrl.value) {
       ctrl.parentNode.replaceChild(ctrl.cloneNode(true), ctrl);
     }
-  }
+ }
+
+function addClipboard (button_id, message) {
+
+    $("#" + button_id).tooltip({
+        trigger : 'hover',
+      })
+      
+    document.getElementById(button_id).onclick = function() {
+        var icon_id = button_id + "-icon";
+        navigator.clipboard.writeText(message);
+        document.getElementById(icon_id).src = 'static/clipboard-check.svg';
+        $("#" + button_id)
+        .attr('data-original-title', "Copied!")
+        .tooltip('show');
+        
+        setTimeout(function() {
+            document.getElementById(icon_id).src = 'static/clipboard.svg';
+            $("#" + button_id)
+            .attr('data-original-title', "Copy")
+        }
+        , 2000);
+    };
+}
+  
+function removeCCLogoFromMobileDevices() {
+    // Following code removes the creative commons logo if the website is opened from a mobile browser
+    const isMobile = /Android|webOS|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)
+    if(isMobile) {
+        document.querySelectorAll(".cc-logo").forEach(el => el.remove());
+    }
+}
+
+window.onload = removeCCLogoFromMobileDevices();
 
 /**
  * HTTP contains some helper methods for making AJAX requests
  */
 class HTTP {
 
     /**
-     * Performs an HTTP GET request and returns a promise with the JSON value of the response
+     * Performs an HTTP GET request and returns a promise with the JSON/CSV value of the response
      * @param {string} url The URL of the GET request
      * @param {number} timeout The timeout, in seconds, of the GET request
      * @param {function} onprogress A function to be called when the request progress information is updated
+     * @param {boolean} csv_file Whether the response is a CSV file
      * @returns {Promise} A promise to the HTTP response
      */
-    static get(url, timeout=null, onprogress=null) {
+    static get(url, timeout=null, onprogress=null, csv_file=false) {
```

Comment:
```suggestion
    static get(url, timeout=null, onprogress=null, parse_json=true) {
```
Function arguments should describe function behavior, not the intended use case of the calling function